### ISSUE TITLE
Subtraction Failure (4 - 2)

### ISSUE THREAD

**Issue by aarongraham9** [jeanlego/rtl_number_lib#3]
Is subtraction supported?

Non interactive mode through the `./libs/librtlnumber/verify_librtlnumber.sh` script gives:
```
INVALID INPUT BITSTRING INDEX: (2)@V_ADD::793
 -X- FAILED == Basic-Subtraction
Basic-Subtraction: Called: ./libs/librtlnumber/rtl_number 2 4 - 2:
Basic-Subtraction: Expected: "2'sb010"
Basic-Subtraction: Got: "arithmetic("4", "-", "2")
pre-processing input: 4
pre-processing input: 2"
```
Causes Code Dump.

Interactivley I get:
```
./libs/librtlnumber/rtl_number

Interactive Mode:
insert operation number of operand [1,2,3] (Q/q to quit): 2
4 - 2
Result is: pre-processing input: 4
pre-processing input: 2
INVALID INPUT BITSTRING INDEX: (2)@V_ADD::793
[1]    14843 abort (core dumped)  ./libs/librtlnumber/rtl_number
```

I haven't had a chance to take a look in code or GDB to see what's going on.

**Comment by jeanlego** [comment]
Yeah subtraction is supported. I just piggy backed on addition and use twos complement to do it. So my twos complement is broken I think... Because division makes use of subtraction to count x number of decrement

Most of the bugs stem from the addition operations being broken I think... Did you test the bitwise and logical operators?? Because under the hood, the addition is making heavy use of them and if it's broken that's where it would stem from.

**Comment by aarongraham9** [comment]
Fixed as of 1e4c2a0a659a93cec34b22e9ad710bb385eb5244.
